components/tools/OmeroPy/test/integration/gatewaytest/test_rdefs.py

```diff
@@ -353,3 +353,106 @@ def testResetDefaults(self, gatewaywrapper):
         # Try resetting (save=True will be ignored, but reset will work)
         i.resetDefaults(save=True)
         assert i.isGreyscaleRenderingModel()
+
+    def testQuantizationSettings(self, gatewaywrapper):
+        """
+        Tests whether quantization settings are properly applied
+        """
+        self.image = gatewaywrapper.getTestImage()
+        channels = self.image.getChannels()
+
+        # collect initial values
+        settings = []
+        for chan in channels:
+            settings.append({
+                "family": chan.getFamily().getValue(),
+                "coefficient": chan.getCoefficient()
+            })
+
+        # channel bounds exceeded check
+        self.image.setQuantizationMap(-1, "logarithmic", 0.1)
+        self.image.setQuantizationMap(200, "exponential", 0.5)
+        # input checks for family (checking against initial values)
+        self.image.setQuantizationMap(0, None, 1)
+        assert settings[0] == {
+            "family": channels[0].getFamily().getValue(),
+            "coefficient": channels[0].getCoefficient()
+        }
+        self.image.setQuantizationMap(0, "no_good_family", 1)
+        assert settings[0] == {
+            "family": channels[0].getFamily().getValue(),
+            "coefficient": channels[0].getCoefficient()
+        }
+        # input checks for coefficient (checking against initial values)
+        self.image.setQuantizationMap(1, "linear", "coefficient")
+        self.image.setQuantizationMap(1, "linear", 100)
+        assert settings[1] == {
+            "family": channels[1].getFamily().getValue(),
+            "coefficient": channels[1].getCoefficient()
+        }
```

Comment:
This ```assert``` and the 2 above it still pass even if I remove all 6 calls to ```self.image.setQuantizationMap()```, so I'm not sure what this is testing, other than ```setQuantizationMap()``` doesn't throw an Exception with invalid inputs?

Reply:
yes, that is the check. that the original values have not been altered. 
see comment: # input checks for family (checking against initial values)